### PR TITLE
Contacts backup

### PR DESCRIPTION
**Pull request by Merkost**
Changes for contacts backup feature

### CODE REVIEW

File: commons/src/main/kotlin/com/simplemobiletools/commons/helpers/BaseConfig.kt [hunk 1]
```diff
@@ -549,4 +549,13 @@ open class BaseConfig(val context: Context) {
...
         get() = prefs.getBoolean(FAVORITES_CUSTOM_ORDER_SELECTED, false)
         set(selected) = prefs.edit().putBoolean(FAVORITES_CUSTOM_ORDER_SELECTED, selected).apply()
+
+    var viewType: Int
+        get() = prefs.getInt(VIEW_TYPE, VIEW_TYPE_LIST)
+        set(viewType) = prefs.edit().putInt(VIEW_TYPE, viewType).apply()
+
+    var contactsGridColumnCnt: Int
+        get() = prefs.getInt(CONTACTS_GRID_COLUMN_COUNT, CONTACTS_GRID_DEFAULT_COLUMNS_COUNT)
+        set(gridLayoutSpanCount) = prefs.edit().putInt(CONTACTS_GRID_COLUMN_COUNT, gridLayoutSpanCount).apply()
```
Comment (tibbi) [inline commons/src/main/kotlin/com/simplemobiletools/commons/helpers/BaseConfig.kt:559]: same thing, keep variable names consistent

File: commons/src/main/kotlin/com/simplemobiletools/commons/helpers/Constants.kt [hunk 2]
```diff
@@ -185,6 +185,13 @@ const val SHOW_PRIVATE_CONTACTS = "show_private_contacts"
 const val MERGE_DUPLICATE_CONTACTS = "merge_duplicate_contacts"
 const val FAVORITES_CONTACTS_ORDER = "favorites_contacts_order"
 const val FAVORITES_CUSTOM_ORDER_SELECTED = "favorites_custom_order_selected"
+const val VIEW_TYPE = "view_type"
+const val CONTACTS_GRID_COLUMN_COUNT = "favourites_column_count"
```
Comment (tibbi) [inline commons/src/main/kotlin/com/simplemobiletools/commons/helpers/Constants.kt:189]: make constant and actual string mean the same thing

File: commons/src/main/kotlin/com/simplemobiletools/commons/helpers/Constants.kt [hunk 3]
```diff
@@ -185,6 +185,13 @@ const val SHOW_PRIVATE_CONTACTS = "show_private_contacts"
...
+
+// contact grid view constants
+const val CONTACTS_GRID_DEFAULT_COLUMNS_COUNT = 3
```
Comment (tibbi) [inline commons/src/main/kotlin/com/simplemobiletools/commons/helpers/Constants.kt:192]: default grid count should depend on the screen size, make it more flexible. Check how is it done in the file manager or other apps.

**Comment by tibbi** [comment]
default column count should come from some resource int, not from code. Check Simple App Launcher or so.

**Comment by Merkost** [comment]
Ready for merging @tibbi

**Comment by tibbi** [comment]
checked just the code now, will check real functionality in the app itself, thanks

**Pull request merged**